File: src/lib/api/plans.ts
import { invoke } from "@tauri-apps/api/core";
import type { Plan } from "./types";

// ── Plans ────────────────────────────────────────────────────────────────

export async function getPlans(featureId: string): Promise<Plan[]> {
  return invoke<Plan[]>("get_plans", { featureId });
}

export async function getPlan(id: string): Promise<Plan> {
  return invoke<Plan>("get_plan", { id });
}

export async function resolvePlan(id: string, status: string, feedback?: string | null): Promise<Plan> {
  return invoke<Plan>("resolve_plan", { id, status, feedback: feedback ?? null });
}

export async function deletePlan(id: string): Promise<void> {
  return invoke<void>("delete_plan", { id });
}

File: src-tauri/src/commands/plans.rs
use tauri::State;

use crate::db;
use crate::AppState;

#[tauri::command]
pub fn get_plans(
    state: State<'_, AppState>,
    feature_id: String,
) -> Result<Vec<db::plans::Plan>, String> {
    let conn = state.db.lock().map_err(|e| e.to_string())?;
    db::plans::get_plans(&conn, &feature_id)
}

#[tauri::command]
pub fn get_plan(state: State<'_, AppState>, id: String) -> Result<db::plans::Plan, String> {
    let conn = state.db.lock().map_err(|e| e.to_string())?;
    db::plans::get_plan(&conn, &id)
}

#[tauri::command]
pub fn resolve_plan(
    state: State<'_, AppState>,
    id: String,
    status: String,
    feedback: Option<String>,
) -> Result<db::plans::Plan, String> {
    let conn = state.db.lock().map_err(|e| e.to_string())?;
    db::plans::resolve_plan(&conn, &id, &status, feedback.as_deref())
}

#[tauri::command]
pub fn delete_plan(state: State<'_, AppState>, id: String) -> Result<(), String> {
    let conn = state.db.lock().map_err(|e| e.to_string())?;
    db::plans::delete_plan(&conn, &id)
}
